learn/user-guide/testing-ballerina-code/writing-tests.md

```diff
@@ -665,6 +665,60 @@ variables can be provided in a file named `Config.toml` located in the tests dir
 For information on using configurable variables, see
  [Configurable Variables](/learn/by-example/configurable.html).
 
+Dependencies meant to be resolved only during testing can be specified in the `Ballerina.toml` file
+by specifiying the scope
+
+```toml
+[[platform.java11.dependency]]
+path = "/user/foo/libs/abc.jar"
+scope = "testOnly"
+```
+
+## Testing Ballerina Services and Clients
+
+Testing Ballerina services involves sending specific requests to the service using a client, 
+and verifying the responses using the assertion functions. The aim is to make sure that the service 
+and client behave as expected when sending and recieving both expected requests and malformed ones.
+
+### Configuring Services and Clients
+
+Services or Clients can be defined in such a way as to accommodate for testing under certain 
+conditions by making use of Configurable variables. Maintaining a test configurable configuration 
+allows us to re-define the host or port as required. For example, when defining a Service, the value
+for the host or port can be specified in the `Config.toml` file, which will be used specifically 
+when running the tests. 
```

Comment:
Let's add a sample Config.toml and mention the path where it needs to be added.

Reply:
We can provide an example service in the section `Testing Services` and make use of the same example to demonstrate test specific Config.toml. We can reorder the topics to do that. WDYT?

Reply:
Added an example and moved the configuration portion towards the end of the guide